server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -835,14 +833,15 @@ private void unassignDeadTablet(TabletLists tLists, TabletMetadata tm) throws Wa
   private void hostUnassignedTablet(TabletLists tLists, KeyExtent tablet,
       UnassignedTablet unassignedTablet) {
     // maybe it's a finishing migration
-    TServerInstance dest = manager.migrations.get(tablet);
+    TServerInstance dest =
+        manager.getContext().getAmple().readTablet(tablet, MIGRATION).getMigration();
     if (dest != null) {
       // if destination is still good, assign it
       if (tLists.destinations.containsKey(dest)) {
         tLists.assignments.add(new Assignment(tablet, dest, unassignedTablet.getLastLocation()));
       } else {
         // get rid of this migration
-        manager.migrations.remove(tablet);
+        manager.getContext().getAmple().mutateTablet(tablet).deleteMigration().mutate();
```

Comment:
This can probably be removed, will comment in more depths elsewhere.  Thinking when we set the future location that would be a natural place to remove any migrations.

Reply:
Dropped in 0a9455cdea88f167179953a1eb8073bf7a77c2f3